curricula/tests/test_user_permissions.py

```diff
@@ -0,0 +1,353 @@
+
+    def test_partner_unit_is_accessible_by_partner(self):
+        request = self.client.get('/partner-curriculum/partner-unit/')
+        request.user = self.partner_user
+        self.assertTrue(self.partner_unit.can_access(request))
+
+    def test_partner_unit_is_accessible_by_author(self):
+        request = self.client.get('/partner-curriculum/partner-unit/')
+        request.user = self.author_user
+        self.assertTrue(self.partner_unit.can_access(request))
+
+    def test_author_unit_is_not_accessible_by_partner(self):
+        request = self.client.get('/author-curriculum/author-unit/')
+        request.user = self.partner_user
+        self.assertFalse(self.author_unit.can_access(request))
+
+    def test_author_unit_is_accessible_by_author(self):
+        request = self.client.get('/author-curriculum/author-unit/')
+        request.user = self.author_user
+        self.assertTrue(self.author_unit.can_access(request))
+
+    def test_partner_lesson_is_accessible_by_partner(self):
+        request = self.client.get('/partner-curriculum/partner-unit/1')
+        request.user = self.partner_user
+        self.assertTrue(self.partner_lesson.can_access(request))
+
+    def test_partner_lesson_is_accessible_by_author(self):
+        request = self.client.get('/partner-curriculum/partner-unit/1')
+        request.user = self.author_user
+        self.assertTrue(self.partner_lesson.can_access(request))
+
+    def test_author_lesson_is_not_accessible_by_partner(self):
+        request = self.client.get('/author-curriculum/author-unit/1')
+        request.user = self.partner_user
+        self.assertFalse(self.author_lesson.can_access(request))
+
+    def test_author_lesson_is_accessible_by_author(self):
+        request = self.client.get('/author-curriculum/author-unit/1')
+        request.user = self.author_user
+        self.assertTrue(self.author_lesson.can_access(request))
+
+    def test_partner_chapter_is_accessible_by_partner(self):
+        request = self.client.get('/partner-curriculum/partner-unit/ch1')
+        request.user = self.partner_user
+        self.assertTrue(self.partner_chapter.is_editable(request))
+
+    def test_partner_chapter_is_accessible_by_author(self):
+        request = self.client.get('/partner-curriculum/partner-unit/ch1')
+        request.user = self.author_user
+        self.assertTrue(self.partner_chapter.is_editable(request))
+
+    def test_author_chapter_is_not_accessible_by_partner(self):
+        request = self.client.get('/author-curriculum/author-unit/ch1')
+        request.user = self.partner_user
+        self.assertFalse(self.author_chapter.is_editable(request))
+
+    def test_author_chapter_is_accessible_by_author(self):
+        request = self.client.get('/author-curriculum/author-unit/ch1')
+        request.user = self.author_user
+        self.assertTrue(self.author_chapter.is_editable(request))
+
+    def create_author_group(self):
```

Comment:
@davidsbailey thoughts on a better way to do this or a way to make it easier for other tests to use this in the future?

Reply:
I would love to make it a factory but was getting a little stuck on how to do it